db/write_batch.cc

```diff
@@ -1602,7 +1606,22 @@ class MemTableInserter : public WriteBatch::Handler {
           cfd->mem()->MarkFlushScheduled()) {
         // MarkFlushScheduled only returns true if we are the one that
         // should take action, so no need to dedup further
-        flush_scheduler_->ScheduleFlush(cfd);
+        flush_scheduler_->ScheduleWork(cfd);
+      }
+    }
+    // check if memtable_list size exceeds max_write_buffer_size_to_maintain
+    if (trim_history_scheduler_ != nullptr) {
+      auto* cfd = cf_mems_->current();
+      assert(cfd != nullptr);
+      if (cfd->ioptions()->max_write_buffer_size_to_maintain > 0 &&
+          cfd->imm()->imm_trim_needed.load(std::memory_order_relaxed) ==
+              false &&
```

Comment:
Why do we move from CAS to a relaxed check? This will cause multiple writers to put the same CF to trim_history_scheduler_.

Reply:
removed since MarkTrimHistoryNeeded() does the same thing